qcodes/dataset/measurements.py

```diff
@@ -625,22 +626,29 @@ def register_parameter(
             basis: The parameters that this parameter is inferred from. If
                 this parameter is not inferred from any other parameters,
                 this should be left blank.
-            paramtype: type of the parameter, i.e. the SQL storage class
+            paramtype: type of the parameter, i.e. the SQL storage class,
+                If None the paramtype will be inferred from the paramter type
+                and the validator.
```

Comment:
I would propose to write "and the validator of the supplied parameter" to be fool-proof.